OpenRA.Mods.D2k/UtilityCommands/D2kMapImporter.cs

```diff
@@ -0,0 +1,419 @@
+			{ 705, Pair.New("RADARO", "Mercenaries") },           //// Mercenary outpost
+			{ 707, Pair.New("LIGHTO", "Mercenaries") },           //// Mercenary light factory
+			{ 708, Pair.New("SILOO", "Mercenaries") },            //// Mercenary silo
+			{ 709, Pair.New("HEAVYO", "Mercenaries") },           //// Mercenary heavy factory
+			{ 710, Pair.New("REPAIRO", "Mercenaries") },          //// Mercenary repair pad
+			{ 711, Pair.New("GUNTOWERO", "Mercenaries") },        //// Mercenary gun turret
+			{ 712, Pair.New("HIGHTECHO", "Mercenaries") },        //// Mercenary high tech factory
+			{ 713, Pair.New("ROCKETTOWERO", "Mercenaries") },     //// Mercenary rocket turret
+			{ 714, Pair.New("RESEARCHO", "Mercenaries") },        //// Mercenary Ix research center
+			{ 715, Pair.New("STARPORTO", "Mercenaries") },        //// Mercenary starport
+			{ 716, Pair.New("PALACEO", "Mercenaries") },          //// Mercenary palace
+			{ 717, Pair.New("RIFLE", "Mercenaries") },            //// Mercenary light inf.
+			{ 718, Pair.New("BAZOOKA", "Mercenaries") },          //// Mercenary trooper
+			{ 719, Pair.New("SABOTEUR", "Mercenaries") },         //// Mercenary saboteur
+			{ 720, Pair.New("HARVESTER", "Mercenaries") },        //// Mercenary engineer
+			{ 721, Pair.New("HARVESTER", "Mercenaries") },        //// Mercenary harvester
+			{ 722, Pair.New("MCVO", "Mercenaries") },             //// Mercenary MCV
+			{ 723, Pair.New("TRIKE", "Mercenaries") },            //// Mercenary trike
+			{ 724, Pair.New("QUAD", "Mercenaries") },             //// Mercenary quad
+			{ 725, Pair.New("COMBATO", "Mercenaries") },          //// Mercenary combat tank
+			{ 726, Pair.New("MISSILETANK", "Mercenaries") },      //// Mercenary missile tank
+			{ 727, Pair.New("SIEGETANK", "Mercenaries") },        //// Mercenary siege tank
+			{ 728, Pair.New("CARRYALLO", "Mercenaries") },        //// Mercenary carryal
+		};
+
+		Map map;
+		Size mapSize;
+		Ruleset rules;
+		TileSet tileSet;
+		string tilesetName;
+		FileStream stream;
+		List<TerrainTile> terrainTiles;
+		Dictionary<int, int> specialValueForTileIndex;
+		List<TerrainTemplateInfo> tileSetsFromYaml;
+
+		D2kMapImporter(string filename, string tileset, Ruleset rules)
+		{
+			tilesetName = tileset;
+			this.rules = rules;
+
+			try
+			{
+				stream = File.OpenRead(filename);
+
+				if (stream.Length == 0 || stream.Length % 4 != 0)
+					throw new Exception("The map is in an unrecognized format!");
+
+				Initialize(filename);
+				LoadMapInfo();
+				FillMap();
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine(e);
+				map = null;
+			}
+			finally
+			{
+				stream.Close();
+			}
+		}
+
+		public static Map Import(string filename, string mod, string tileset, Ruleset rules)
+		{
+			var map = new D2kMapImporter(filename, tileset, rules).map;
+			if (map == null)
+				return null;
+
+			map.RequiresMod = mod;
+			map.MakeDefaultPlayers();
+
+			// TODO: Address FixOpenAreas crash
```

Comment:
I think it is not needed for Dune 2000 imports which pick individual (pre-randomized) tile images:

![image](https://cloud.githubusercontent.com/assets/756669/6183003/3f562ec6-b34f-11e4-9ba8-a2a69780afb2.png)

This is only a problem in RA/TD when the map import marks terrain as Clear and everything is tile variant 0 which looks wallpaper like:

![image](https://cloud.githubusercontent.com/assets/756669/6183015/739d7036-b34f-11e4-8187-379989945f0f.png)

which gets converted by "Fix Open Areas" to

![image](https://cloud.githubusercontent.com/assets/756669/6183018/81c82be2-b34f-11e4-8bbd-8bb612eb8ce8.png)

I automatized this for both clear tile placement and on new map generation in both the legacy WinForms editor and the upcoming in-game one #4496 for better usability and because it was often forgot to manually hit that button.